components/feature/addons/src/main/java/mozilla/components/feature/addons/update/db/AddonUpdateRequestDao.kt

```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.addons.update.db
+
+import androidx.room.Insert
+import androidx.room.Query
+import androidx.room.Dao
+import androidx.room.OnConflictStrategy
+
+/**
+ * Internal dao for accessing and modifying add-on update requests in the database.
+ */
+@Dao
+internal interface AddonUpdateRequestDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insertOrUpdate(entity: AddonUpdateRequestEntity): Long
+
+    @Query("SELECT * FROM addon_update_requests where addon_id =:addonId LIMIT 1")
```

Comment:
We shouldn't need the `LIMIT 1` here as addon_id is the primary key so this query can only ever return one record anyway?

Reply:
Agree 👍